airflow/utils/timeout.py

```diff
@@ -39,8 +40,9 @@ def handle_timeout(self, signum, frame):
 
     def __enter__(self):
         try:
-            signal.signal(signal.SIGALRM, self.handle_timeout)
-            signal.alarm(self.seconds)
+            if isinstance(threading.current_thread(), threading._MainThread):
```

Comment:
Yeah, no signal handler is installed on the background thread and long loading dags are not killed